### PR TITLE
Cec2020

### PR DESCRIPTION
**Pull request by ErlingWis**
PR contains:

- CMOPs 1-6 from the CEC2020 problem suite.
- Parallelisation of the simulation runs, multiple runs will now run in parallell.
- Restructure of test suites and cmops.
- Plotter that uses references to the pps attributes, functionality for plotting is now outside the PPS struct.
- Filehandling package
- Some functions moved to struct methods.
- Updated example README

### CODE REVIEW

File: pps/pps+private.go [hunk 1]
```diff
@@ -13,11 +10,11 @@ import (
...
 
-func (pps *PPS) setIdealAndNadir(generation int) {
+func (pps *PPS) setIdealAndNadir() {
 	ip, np := biooperators.CalculateNadirAndIdealPoints(pps.moea.Population())
-
-	pps.idealPoints[generation] = ip
-	pps.nadirPoints[generation] = np
+	//Append in stead of initialising arrays at a fixed size to
+	pps.idealPoints = append(pps.idealPoints, ip)
+	pps.nadirPoints = append(pps.nadirPoints, np)
```
Comment (ErlingWis) [inline pps/pps+private.go:17]: Had issues setting a static array size for different population sizes and maxFuncEvals, changed this to append each generation in stead.

File: simulator/simulator.go [hunk 2]
```diff
@@ -2,72 +2,85 @@ package simulator
...
+
 func (s *Simulator) Simulate() {
-	for _, p := range s.Problems {
-		cmop := testSuite.NewLIRCMOP(p)
+	for p, cmop := range s.TestSuite.Problems {
+
+		start := time.Now()
+		fmt.Println("Starting run of", cmop.Name)
+
+		channel := make(chan float64)
```
Comment (mariusaarsnes) [inline simulator/simulator.go:38]: Åløh

File: types/MOEA.go [hunk 3]
```diff
@@ -4,14 +4,15 @@ import (
...
 type CMOP interface {
 	NumberOfObjectives() int
 	NumberOfConstraints() int
 	Name() string
 	Calculate(Genotype) Fitness
 }
-
+*/
 // ConstraintType describes which type of constraint it is. Either a equals-or-less-than or equals-or-greater-than constraint.
 // Should we assume only inequality constraints by using a small delta? Seems like most approaches do
```
Comment (mariusaarsnes) [inline types/MOEA.go]:
```suggestion
```

**Review by mariusaarsnes — APPROVED** [review]
Great work!

**Pull request merged**